### PR TITLE
Created test CI/CD pipeline

### PR DESCRIPTION
**Pull request by nicholaslegrand102**
This PR contains a  Github workflow yml configuration file that is used to create a pipeline that is executed  each time a pull request is opened with main or if a push is directly done to the main branch. It also contains some modifications to standardize the existing tests.

The pipeline is divided in many steps:
 - Check out the the repository
 - Verify if all modules can be built successfully
 - Run all the tests 
 
The pipeline will return an error if either of these steps are terminated unsuccessfully and we use a runner with an Ubuntu image because it is the only Linux based image that they host.

I also added a badge to the README.md that shows the state of the pipeline.

Github does not allow pipeline executions for pipelines being merged from forks so if you wish to see the pipeline execution please see this PR https://github.com/felix642/1chipML/pull/48 in the "Checks" section.

### CODE REVIEW

File: .github/workflows/make.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  run-tests:
+    name: Run tests
+    runs-on: ubuntu-latest
```
Comment (felix642) [inline .github/workflows/make.yml:14]: Are we able to select a specific version? We need the pipeline to be as stable as possible. We don't want it to break if ubuntu decides to release a new version with a new compiler. I think we should use the image "ubuntu-22.04" and update it manually in the future if needed.

File: .github/workflows/make.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        working-directory: .
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v3
+
+      - name: Build files
+        run: make all 
+
+      - name: Run tests
+        run: make run_all_tests
```
Comment (felix642) [inline .github/workflows/make.yml:26]: We will need to adapt the tests that are currently on the repository to return 1 (or an error code) if an error happens. Can you do it on this PR?

File: README.md [hunk 3]
```diff
@@ -8,6 +9,7 @@
   1chipML
 </h1>
 
+![Build and Tests](https://github.com/felix642/1chipML/actions/workflows/make.yml/badge.svg?style=flat)
```
Comment (felix642) [inline README.md:12]: This is cool, but the link is not valid. It should be the main repo, not the fork ;)

**Review by felix642 — CHANGES_REQUESTED** [review]

File: .github/workflows/make.yml (hunk 1 above)
Comment (nicholaslegrand102) [reply .github/workflows/make.yml:14]: yes we are able to select a version

File: README.md (hunk 3 above)
Comment (nicholaslegrand102) [reply README.md:12]: Good catch

File: .github/workflows/make.yml (hunk 2 above)
Comment (nicholaslegrand102) [reply .github/workflows/make.yml:26]: Yes I can take care of standardizing all the tests

**Review by felix642 — APPROVED** [review]
LGTM

File: tests/test_linear_congruential_random_generator.c [hunk 4]
```diff
@@ -15,10 +15,18 @@ int main(void) {
   int i, n = 100;
 
   printf("linear congruential random generator\n");
-  for (i = 0; i < n; i++)
-    printf("%1.3f ", linear_congruential_random_generator());
-  printf("\n");
-  return (0);
+  for (i = 0; i < n; i++) {
```
Comment (florenceCloutier) [inline tests/test_linear_congruential_random_generator.c:18]: I don't get why there is a loop here? Is there a specific reason why it runs 100 times?

**Review by florenceCloutier — CHANGES_REQUESTED** [review]

**Review by bisson2000 — APPROVED** [review]
lgtm

**Pull request merged**